### PR TITLE
Update upgrade command, add RELEASE notes

### PR DESCRIPTION
**Pull request by iameskild**
## Reference Issues or PRs

When I was testing `2023.5.1rc1` by upgrading an existing Nebari cluster, I ran into an issue where the deployment failed and complained about an existing service account:

```sh
[terraform]: │ Error: serviceaccounts "argo-admin" already exists
[terraform]: │
[terraform]: │   with module.argo-workflows[0].kubernetes_service_account_v1.argo-admin-sa,
[terraform]: │   on modules/kubernetes/services/argo-workflows/main.tf line 150, in resource "kubernetes_service_account_v1" "argo-admin-sa":
[terraform]: │  150: resource "kubernetes_service_account_v1" "argo-admin-sa" {
[terraform]: │
[terraform]: ╵
```

Although I was able to redeploy and get around this issue, I think it would be safer to have users delete the existing service accounts before deploying. Given that we are already requiring them to delete the CRDs, this adds a minimal additional step. 

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [x] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: nebari/upgrade.py [hunk 1]
```diff
@@ -398,14 +398,23 @@ def _version_specific_upgrade(
...
...cyan]Note:[/] Upgrading requires a one-time manual deletion of the Argo Workflows Custom Resource Definitions (CRDs) and service accounts. \n\n[red bold]Warning:  [link=https://{config['domain']}/argo/workflows]Workflows[/link] and [link=https://{config['domain']}/argo/workflows]CronWorkflows[/link] created before deleting the CRDs will be erased when the CRDs are deleted and will not be restored.[/red bold] \n\nThe updated CRDs will be installed during the next [cyan bold]nebari deploy[/cyan bold] step. Argo Workflows will not function after deleting the CRDs until the updated CRDs are installed in the next nebari deploy. You must delete the Argo Workflows CRDs and service accounts before upgrading to {self.version} (or later) or deploy step will fail.  Please delete them before proceeding by generating a kubeconfig (see [link=https://www.nebari.dev/docs/how-tos/debug-nebari/#generating-the-kubeconfig]docs[/link]), installing kubectl (see [link=https://www.nebari.dev/docs/how-tos/debug-nebari#installing-kubectl]docs[/link]), and running the following two commands:\n\n\t[cyan bold]{kubectl_delete_argo_crds_cmd} [/cyan bold]\n\n\t[cyan bold]{kubectl_delete_argo_sa_cmd} [/cyan bold]"
```
Comment (Adam-D-Lewis) [inline nebari/upgrade.py:406]: "Argo Workflows will not function after deleting the CRDs until the updated CRDs are installed" I would change "updated CRDs" to "updated CRDs and service accounts"

Comment (Adam-D-Lewis) [reply nebari/upgrade.py:406]: Could you change "or deploy step will fail" to "or the deploy step will fail"?

Comment (iameskild) [reply nebari/upgrade.py]: Done! Thanks :)

File: RELEASE.md [hunk 2]
```diff
@@ -11,6 +11,64 @@ This file is copied to nebari-dev/nebari-docs using a GitHub Action. -->
...
+
+* Upgrade Argo-Workflows to version 3.4.4
+
+### Breaking Changes
+
+* The Argo-Workflows version upgrade will result in a breaking change if the existing Kubernetes CRDs are not deleted (see the NOTE below for more details).
+* There is a minor breaking change for the Nebari CLI version shorthand, previously it `nebari -v` and now to align with Python convention, it will be `nebari -V`.
+
+> NOTE: After installing the Nebari version `2023.5.1`, please run `nebari upgrade -c nebari-config.yaml` to upgrade
+> the `nebari-config.yaml`. This command will also prompt you with delete a few Kubernetes resources (specifically
```
Comment (Adam-D-Lewis) [inline RELEASE.md:26]: "prompt you with delete" -> "prompt you to delete"

**Review by Adam-D-Lewis — APPROVED** [review]

**Pull request merged**